elip-ct-descriptors.mediawiki

```diff
@@ -0,0 +1,151 @@
+* <code>slip77(<64-character hex>)</code> which indicates that blinding keys for these addresses are derived via SLIP77; or
+* <code><KEY></code> which is an ordinary (extended) public key; the blinding key is derived by a Taproot-style tweak of the key with the data of <code><DESCRIPTOR></code>.
+
+In the <code><KEY></code> expression, all key(s) must be compressed; x-only and uncompressed keys are invalid.
+
+It is permissible, and likely to be common, for a <code><KEY></code> key to be a private key (e.g. an xprv) even if the keys in the actual descriptor are public keys.
+Such a descriptor is termed a '''view descriptor''', and the blinding private key a '''view key'''.
+
+===Drawbacks===
+
+* Because there is no sensible way to do multiparty unblinding, the blinding key must be a single key whose private half is known to all participants.
+* The use of "view descriptors" which are nominally public but contain secret key data, may lead to user confusion; though we argue it has similar privacy properties to the chaincode value included in xpubs.
+* Any party who has a copy of an addresses' view descriptor is able to see the blinding key and unblind coins sent to that address, or *any* derived address in the case of a descriptor containing wildcards.
+
+The latter point is the intended use of view descriptors, and is listed as a drawback only because it may be surprising.
+
+===Specification===
+
+First, the <code>ct</code> descriptor is defined as above: its string serialization is given by <code>ct(<BLINDING_KEY>, <DESCRIPTOR>)</code> where <code>DESCRIPTOR</code> is the string serialization of an ordinary descriptor.
+The <code>scriptPubKey</code> corresponding to a <code>ct</code> descriptor is that corresponding to the embedded <code>DESCRIPTOR</code>. The encoding and semantics of <code>BLINDING_KEY</code> are given below.
+
+<code>BLINDING_KEY</code> expressions are one of
+* '''slip77''', encoded as <code>slip77(<64-characters hex>)</code>, whose semantics are that the 64 hex characters are interpreted as the 32-byte <code>master_blinding_key</code> in SLIP77; the <code>scriptPubKey</code> for SLIP77 is computed as normal from the ordinary descriptor.
+** These 32 bytes are the same for both public and private descriptors; they have no "public" equivalent.
+** This mode is not recommended because there is no way to express this descriptor without revealing the SLIP77 key, which can be used to unblind every single output received by the wallet.
+* '''bare key''', which simply encodes a single key in the same way as the keys in <code><DESCRIPTOR></code> are encoded. The blinding key is derived from the address by taking the public key ''K'' and tweaking it to form a new key ''K<sub>blind</sub>'' as follows:
+** ''K' = K + H<sub>tag</sub>(K, <code>scriptPubKey</code>)'', where
```

Comment:
Should this be `H<sub>tag</sub>(K, <code>scriptPubKey</code>) * G`? or `K` is a private key?